### DISCUSSION TITLE
Recognizing functions added by SKSE

### DISCUSSION THREAD

**Discussion by sm644** [blu3mania/npp-papyrus#37] [Q&A] [answered]
This plugin seems to work very nicely within NPP, even without the need for bat files.
However, when I try to compile a psc that contains functions provided by SKSE (GetKeyword, for instance), it will not accept that they are actually functions. I have all the SKSE psc files in the Scripts\Source and Source\Scripts directories, both for good measure.
How do I update the Function List?
Thanks in advance for any guidance you can provide, because I can find very little by googling.

**Comment by blu3mania** [comment]
This plugin just calls the game's PapyrusCompiler.exe to compile scripts, so if it doesn't compile, then your Skyrim setup is not fully working and you will encounter the same problem in CK as well. Since you mentioned problem with GetKeyword, I can only assume that you do not have Keyword.psc in your script source folder.

**Reply by sm644** [reply]
Hello and thanks for the suggestion.
It is compiling ok until I try a script that includes any SKSE functions. It seems to work nicely with original game scripts. I have Keyword.psc in the source folder, but I do wonder if there should be an updated Function List. Does Papyrus.xml or Papyrus Script.xml provide a function list? I have seen talk of a Functionlist.xml, but that might be a disused name now.
In NPP, the function list window shows me all the non-SKSE functions in the loaded psc.

PS. Just to be sure, I have put keyword.psc and all the SKSE scripts in Data\Scripts\Source, Data\Source and Data\Source\Scripts.

**Comment by blu3mania** [comment, marked as answer]
Notepad++'s Function List has nothing to do with PapyrusCompiler. Also, Papyrus.xml is only for styling. The only thing that could affect compilation is the game configuration in this plugin, and especially Import directories. Make sure you have {Skyrim Install Folder}\Data\Scripts\Source there, and without trailing backslash. Though, since you said it works with original game scripts, this should not be an issue.

Does your code compile in CK? Another thing you could try to verify your game setup is to open a command prompt, change directory to {Skyrim Install Folder}\Data\Scripts\Source, then run this command to see if it compiles (replace folder according to your setup):

`"..\..\..\Papyrus Compiler\PapyrusCompiler.exe" {Your Script Source}.psc -i="{Skyrim Install Folder}\Data\scripts\source" -o="{Any Test Folder}" -f=TESV_Papyrus_Flags.flg`

**Reply by sm644** [reply]
Thanks for the command prompt template. That certainly narrows things down.
The prompt works correctly with an existing psc (Activemagiceffect.psc, in this case).
However, _slpnpc.psc, from SleepTight Fixes, gives the stream of errors for not recognizing functions. I think this proves it is not an issue within NPP and your plugin.
Keyword.psc is definitely present, so I will have find out why the compiler doesn't like it, or can't find it.

**Reply by sm644** [reply]
Indeed, when I use the command prompt to compile keyword.psc, that works with no error.

**Reply by sm644** [reply]
Actually, it looks like my only problem is that a couple of commands in a supposedly working script are not correct. Your excellent plugin is blameless. Thanks again.

**Reply by blu3mania** [reply]
You are welcome, and happy to know that you nailed down the problem.
